### PR TITLE
Get `npm install` to work

### LINKED ISSUE #106: We could not find the 'npm' command. Make sure it's in $PATH
I get this notification from subject.

but 

```
$ which npm
/Users/david/n/bin/npm
```

works...

Not sure if it's related, when I add "hyperpower" plugin to `.hyperterm.js` and save it, nothing really happens and 

```
$ ls .hyperterm_plugins/
local        package.json
```

Any suggestions ?

Should "hyperpower" plugin install automatically after saving the file and is this not happening because of the problem about not finding 'npm' command supposedly?

### PR DESCRIPTION
**Pull request by marcbachmann**
This fixes  #106 and #145

### CODE REVIEW

**Comment by montogeek** [comment]
@marcbachmann How did you tested it? I just compiled a new version with your changes and the problem persists

**Comment by montogeek** [comment]
Also, Could you please fix the dependency version?

File: package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "gaze": "1.1.0",
     "mkdirp": "0.5.1",
     "ms": "0.7.1",
-    "uid2": "0.0.3"
+    "uid2": "0.0.3",
+    "user-env": "^2.0.0"
```
Comment (rauchg) [inline package.json]: Can you pin this please?

**Comment by rauchg** [comment]
What an elegant solution! Thank you @marcbachmann. Only one minor nit-pick.

**Comment by marcbachmann** [comment]
Done. Rebased and version fixed.

**Comment by montogeek** [comment]
Marvelous

**Comment by marcbachmann** [comment]
Another solution would be to install `npm` as module and use the api.
That way it's also windows-compatible.

```js
const npm = require('npm')
function install (cb) {
  npm.load({}, function (err) {
    if (err) return cb(err)
    npm.commands.prune([], function (err) {
      if (err) return cb(err)
      npm.commands.install(['--production'], cb)
    })
  })
}
```

**Comment by rauchg** [comment]
@marcbachmann I could be wrong but I've heard that the npm team is not recommending API usage. Is this the case @iarna @othiym23 ?

**Comment by leo** [comment]
@rauchg [It is.](https://www.youtube.com/watch?v=vLcOkHhEt1g&feature=youtu.be&t=21m1s) Like @othiym23 suggests it, we should a child process instead.

**Comment by marcbachmann** [comment]
I've replaced `user-env` with `shell-env` because it also has windows support and has an async function. I've also removed the prefix that wasn't needed anymore.

@rauchg this is ready to merge.

**Comment by rauchg** [comment]
<3

**Comment by mjgyesme** [comment]
woo hoo! nvm users rejoice!

**Comment by iarna** [comment]
Indeed! We would prefer that you not use npm's internal API, we'd like to keep it subject to change. (Mostly because it's terrible, and we'd like to have the freedom to make it less terrible.)

Calling npm is the correct and supported way of integrating with npm.

If you find yourself having trouble integrating with it (eg it's output is insufficiently uniform, say) then I'd be happy to work with you all in patching it to correct that sort of thing.

**Comment by rauchg** [comment]
Thanks a lot for letting us know @iarna

**Pull request merged**